engine/Assets/Scripts/Analytics/AnalyticsManager.cs

```diff
@@ -0,0 +1,145 @@
+// #define DEBUG_ANALYTICS // Uncomment this line to print analytic debug information
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Synthesis.PreferenceManager;
+using Unity.Services.Analytics;
+using Unity.Services.Core;
+using UnityEngine;
+
+namespace Analytics {
+    // TODO: after testing, disable when in editor
+
+    /// <summary>
+    /// Handles unity analytics initialization and sending custom events.
+    /// View analytics on the <a href = "https://dashboard.unity3d.com/">Unity Dashboard.</a>
+    /// Events will be be sent if the application is running in the editor
+    /// </summary>
+    public class AnalyticsManager : MonoBehaviour {
+        public const string USE_ANALYTICS_PREF = "analytics/use_analytics";
+
+        private static bool _useAnalytics;
+
+        private async void Start() {
+            _useAnalytics = PreferenceManager.GetPreference<bool>(USE_ANALYTICS_PREF);
+
+            SynthesisAPI.EventBus.EventBus.NewTypeListener<PostPreferenceSaveEvent>(e => {
+                bool useAnalytics = PreferenceManager.GetPreference<bool>(USE_ANALYTICS_PREF);
+
+                if (useAnalytics == _useAnalytics)
+                    return;
+
+                _useAnalytics = useAnalytics;
+
+                if (useAnalytics)
+                    StartDataCollection();
+                else
+                    StopDataCollection();
+            });
+
+            if (!Application.isEditor)
+                await UnityServices.InitializeAsync();
+
+#if DEBUG_ANALYTICS
+            AnalyticsDebug("Unity services initialized");
+#endif
+
+            // TODO: check if the user gives consent to collect information
```

Comment:
Should we make this part of Hunter tracking down multiplayer legal forms?

Reply:
Yeah I think so. If we need to, we can add a popup at the beginning to ask new users.

Reply:
👍